async-nats/benches/jetstream.rs

```diff
@@ -0,0 +1,215 @@
+                            name: "bench".to_owned(),
+                            subjects: vec!["bench".to_string()],
+                            ..Default::default()
+                        })
+                        .await
+                        .unwrap();
+                    stream.purge().await.unwrap();
+                    context
+                });
+
+                b.to_async(rt).iter_with_large_drop(move || {
+                    let nc = context.clone();
+                    async move {
+                        publish_async_batch(nc, Bytes::from_static(&MSG[..size]), messages_per_iter)
+                            .await
+                    }
+                });
+            },
+        );
+    }
+    throughput_group.finish();
+
+    let mut messages_group = c.benchmark_group("jetstream async publish messages amount");
+    messages_group.sample_size(10);
+    messages_group.warm_up_time(std::time::Duration::from_secs(1));
+
+    for &size in [32, 1024, 8192].iter() {
+        messages_group.throughput(criterion::Throughput::Elements(messages_per_iter));
+        messages_group.bench_with_input(
+            criterion::BenchmarkId::from_parameter(size),
+            &size,
+            |b, _| {
+                let rt = tokio::runtime::Runtime::new().unwrap();
+                let context = rt.block_on(async {
+                    let context = async_nats::jetstream::new(
+                        async_nats::connect(server.client_url()).await.unwrap(),
+                    );
+
+                    let stream = context
+                        .create_stream(stream::Config {
+                            name: "bench".to_owned(),
+                            subjects: vec!["bench".to_string()],
+                            ..Default::default()
+                        })
+                        .await
+                        .unwrap();
+                    stream.purge().await.unwrap();
+                    context
+                });
+
+                b.to_async(rt).iter_with_large_drop(move || {
+                    let context = context.clone();
+                    async move {
+                        publish_async_batch(
+                            context,
+                            Bytes::from_static(&MSG[..size]),
+                            messages_per_iter,
+                        )
+                        .await
+                    }
+                });
+            },
+        );
+    }
+    messages_group.finish();
+}
+async fn publish_sync_batch(context: async_nats::jetstream::Context, msg: Bytes, amount: u64) {
+    for _i in 0..amount {
+        context
+            .publish("bench".into(), msg.clone())
+            .await
+            .unwrap()
+            .await
+            .unwrap();
+    }
+}
+
+async fn publish_async_batch(context: async_nats::jetstream::Context, msg: Bytes, amount: u64) {
+    // This acts as a semaphore that does not allow for more than 1000 publish acks awaiting.
```

Comment:
10? not 10k?

Reply:
We get better perf with 10. Updated comment.